app/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDao.java

```diff
@@ -88,39 +95,100 @@ public SchedulePlan getSchedulePlan(StudyIdentifier studyIdentifier, String guid
     public SchedulePlan createSchedulePlan(StudyIdentifier studyIdentifier, SchedulePlan plan) {
-        checkNotNull(studyIdentifier, "StudyIdentifier is null");
-        checkNotNull(plan, "Schedule plan is null");
+        checkNotNull(studyIdentifier);
+        checkNotNull(plan);
         
         plan.setStudyKey(studyIdentifier.getIdentifier());
         plan.setGuid(BridgeUtils.generateGuid());
         plan.setModifiedOn(DateUtils.getCurrentMillisFromEpoch());
+        
+        forEachCriteria(plan, scheduleCriteria -> persistCriteria(scheduleCriteria));
         mapper.save(plan);
         return plan;
     }
 
     @Override
     public SchedulePlan updateSchedulePlan(StudyIdentifier studyIdentifier, SchedulePlan plan) {
-        checkNotNull(studyIdentifier, "StudyIdentifier is null");
-        checkNotNull(plan, "Schedule plan is null");
+        checkNotNull(studyIdentifier);
+        checkNotNull(plan);
         
         plan.setStudyKey(studyIdentifier.getIdentifier());
         plan.setModifiedOn(DateUtils.getCurrentMillisFromEpoch());
+        
+        forEachCriteria(plan, scheduleCriteria -> persistCriteria(scheduleCriteria));
         mapper.save(plan);
         return plan;
     }
 
     @Override
     public void deleteSchedulePlan(StudyIdentifier studyIdentifier, String guid) {
-        checkNotNull(studyIdentifier, "Study identifier is null");
-        checkArgument(StringUtils.isNotBlank(guid), "Plan GUID is blank or null");
+        checkNotNull(studyIdentifier);
+        checkArgument(isNotBlank(guid));
         
         SchedulePlan plan = getSchedulePlan(studyIdentifier, guid);
+        
+        forEachCriteria(plan, scheduleCriteria -> deleteCriteria(scheduleCriteria));
         mapper.delete(plan);
     }
+    
+    private void forEachCriteria(SchedulePlan plan, Function<ScheduleCriteria,Criteria> consumer) {
+        if (plan.getStrategy() instanceof CriteriaScheduleStrategy) {
+            CriteriaScheduleStrategy strategy = (CriteriaScheduleStrategy)plan.getStrategy();
+            for (int i=0; i < strategy.getScheduleCriteria().size(); i++) {
+                ScheduleCriteria scheduleCriteria = strategy.getScheduleCriteria().get(i);
+                
+                // Add the key... builder makes this verbose
+                scheduleCriteria = new ScheduleCriteria.Builder()
+                        .withSchedule(scheduleCriteria.getSchedule())
+                        .withCriteria(scheduleCriteria.getCriteria())
+                        .withKey(getKey(plan, i)).build();
+
+                Criteria criteria = consumer.apply(scheduleCriteria);
+
+                // Now add the new criteria object and set new object in list
+                scheduleCriteria = new ScheduleCriteria.Builder()
+                        .withSchedule(scheduleCriteria.getSchedule())
+                        .withCriteria(criteria).build();
+                strategy.getScheduleCriteria().set(i, scheduleCriteria);
```

Comment:
The save and the delete don't really need this block, since we don't really need to save criteria into the SchedulePlans table, and delete will just ignore it anyway. It might be cleaner to just have separate loops and helper methods for get, save, and delete.